x/concentrated-liquidity/swaps_test.go

```diff
@@ -1117,69 +791,311 @@ func (s *KeeperTestSuite) TestCalcAndSwapInAmtGivenOut() {
 			tokenInDenom:               "usdc",
 			priceLimit:                 sdk.NewDec(4994),
+			swapFee:                    sdk.ZeroDec(),
 			expectedTokenOut:           sdk.NewCoin("eth", sdk.NewInt(12891)),
 			expectedTokenIn:            sdk.NewCoin("usdc", sdk.NewInt(64417624)),
 			expectedTick:               sdk.NewInt(309941),
 			expectedLowerTickFeeGrowth: DefaultFeeAccumCoins,
 			expectedUpperTickFeeGrowth: DefaultFeeAccumCoins,
 		},
+	}
+	swapInGivenOutErrorTestCases = map[string]SwapTest{
 		"single position within one tick, trade does not complete due to lack of liquidity: usdc -> eth": {
-			addPositions: func(ctx sdk.Context, poolId uint64) {
-				// add first position
-				_, _, _, err := s.App.ConcentratedLiquidityKeeper.CreatePosition(ctx, poolId, s.TestAccs[0], DefaultAmt0, DefaultAmt1, sdk.ZeroInt(), sdk.ZeroInt(), DefaultLowerTick, DefaultUpperTick)
-				s.Require().NoError(err)
-			},
 			tokenOut:     sdk.NewCoin("usdc", sdk.NewInt(5300000000)),
 			tokenInDenom: "eth",
 			priceLimit:   sdk.NewDec(6000),
+			swapFee:      sdk.ZeroDec(),
 			expectErr:    true,
 		},
 		"single position within one tick, trade does not complete due to lack of liquidity: eth -> usdc": {
-			addPositions: func(ctx sdk.Context, poolId uint64) {
-				// add first position
-				_, _, _, err := s.App.ConcentratedLiquidityKeeper.CreatePosition(ctx, poolId, s.TestAccs[0], DefaultAmt0, DefaultAmt1, sdk.ZeroInt(), sdk.ZeroInt(), DefaultLowerTick, DefaultUpperTick)
-				s.Require().NoError(err)
-			},
 			tokenOut:     sdk.NewCoin("eth", sdk.NewInt(1100000)),
 			tokenInDenom: "usdc",
 			priceLimit:   sdk.NewDec(4000),
+			swapFee:      sdk.ZeroDec(),
 			expectErr:    true,
 		},
 	}
+)
+
+func (s *KeeperTestSuite) TestCalcAndSwapOutAmtGivenIn() {
+	additiveFeeGrowthGlobalErrTolerance := osmomath.ErrTolerance{
+		// 2 * 10^-18
+		AdditiveTolerance: sdk.SmallestDec().Mul(sdk.NewDec(2)),
+		// actual fee growth should be greater than expected.
+		RoundingDir: osmomath.RoundUp,
+	}
+
+	tests := make(map[string]SwapTest, len(swapOutGivenInCases)+len(swapOutGivenInFeeCases)+len(swapOutGivenInErrorCases))
+	for name, test := range swapOutGivenInCases {
+		tests[name] = test
+	}
+
+	for name, test := range swapOutGivenInFeeCases {
+		tests[name] = test
+	}
+
+	// add error cases as well
+	for name, test := range swapOutGivenInErrorCases {
+		tests[name] = test
+	}
 
 	for name, test := range tests {
+		test := test
 		s.Run(name, func() {
 			s.Setup()
 			s.FundAcc(s.TestAccs[0], sdk.NewCoins(sdk.NewCoin("eth", sdk.NewInt(10000000000000)), sdk.NewCoin("usdc", sdk.NewInt(1000000000000))))
 			s.FundAcc(s.TestAccs[1], sdk.NewCoins(sdk.NewCoin("eth", sdk.NewInt(10000000000000)), sdk.NewCoin("usdc", sdk.NewInt(1000000000000))))
 
 			// Create default CL pool
-			pool := s.PrepareConcentratedPool()
+			pool := s.PrepareCustomConcentratedPool(s.TestAccs[0], ETH, USDC, DefaultTickSpacing, DefaultExponentAtPriceOne, sdk.ZeroDec())
```

Comment:
Was the custom CL pool func needed here to specify the testAcc?

Reply:
No, will change back in the next PR

Reply:
fixed now